src/API/Google/Middleware.php

```diff
@@ -613,4 +618,22 @@ public function account_request_review( $regions ) {
 			);
 		}
 	}
+
+	/**
+	 * This function detects if the current account is a sub-account
+	 *
+	 * @return bool True if it's a standalone account.
+	 */
+	public function is_subaccount(): bool {
+		$merchant_id = $this->options->get_merchant_id();
+		$accounts    = $this->get_merchant_accounts();
+
+		foreach ( $accounts as $account ) {
+			if ( $account['id'] === $merchant_id && $account['subaccount'] ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
```

Comment:
The only thing I don't like about this approach is that the WCS turns this into several requests (one for account list + one per account for details). I might be missing something, but as far as I can see this is requested every time we view the Product Feed page, since it isn't cached.

Can we cache this in a transient to prevent it from being requested repeatedly. As far as I'm aware an account can't change from sub account to standalone, at least not easily. So repeatedly asking the account type isn't necessary.
Or do you think it's worth it to have a separate endpoint in WCS to get the sub-account status. Or even rewrite the get merchants endpoint to flag whether we want account detail requests to be sent.

Reply:
> I might be missing something, but as far as I can see this is requested every time we view the Product Feed page, since it isn't cached.

This is cached  for 20 mins on `get_review_status` function inside `RquestReviewController`

I can do as well another cache only for the standalone function without expiration time.

That is done here https://github.com/woocommerce/google-listings-and-ads/pull/1555/commits/ff14ff68afe26612b665cc0fe7875098b9049ecd